src/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -148,8 +148,10 @@ public RestoreCommand()
         private async Task<bool> PerformNuGetV3RestoreAsync(string packagesDir, string inputPath)
         {
             var inputFileName = Path.GetFileName(inputPath);
-            PackageSpec packageSpec;
+            var projectDirectory = Path.GetDirectoryName(Path.GetFullPath(inputPath));
```

Comment:
We made fixes to this class around full path to happen right off the bat in ExecuteCommandAsync() when there are arguments and in DetermineRestoreInputs() method when there no explicit arguments. I would move all the Path.GetFullPath() operation to that method if it is not done already. This makes the code easier to understand